test/algorithms/set_operations/set_ops_areal_areal.cpp

```diff
@@ -0,0 +1,218 @@
+
+#include <boost/geometry.hpp>
+
+#include <algorithms/overlay/overlay_cases.hpp>
+#include <algorithms/overlay/multi_overlay_cases.hpp>
+
+#include <boost/test/included/test_exec_monitor.hpp>
+
+// Use another alias, on purpose it is not "bg::" here
+namespace bgeo = boost::geometry;
+
+struct ut_settings
+{
+    explicit ut_settings()
+    {}
+
+    inline ut_settings& ignore_validity()
+    {
+        test_validity = false;
+        return *this;
+    }
+
+    inline ut_settings& ignore_validity_diff()
+    {
+        test_validity_diff1 = false;
+        test_validity_diff2 = false;
+        test_validity_diff_sym = false;
+        return *this;
+    }
+
+    inline ut_settings& ignore_validity_diff1()
+    {
+        test_validity_diff1 = false;
+        return *this;
+    }
+
+    inline ut_settings& ignore_validity_diff2()
+    {
+        test_validity_diff2 = false;
+        return *this;
+    }
+
+    inline ut_settings& ignore_validity_diff_sym()
+    {
+        test_validity_diff_sym = false;
+        return *this;
+    }
+
+    inline ut_settings& set_epsilon(long double epsilon)
+    {
+        epsilon = epsilon;
+        return *this;
+    }
+
+    bool test_validity{true};
+    bool test_validity_diff1{true};
+    bool test_validity_diff2{true};
+    bool test_validity_diff_sym{true};
+
+    long double epsilon{1.0e-6};
+};
+template <typename T, typename Multi, typename Geometry1, typename Geometry2>
+void test_detail(std::string const& name, std::string const& wkt1, std::string const& wkt2,
+    ut_settings const& settings)
+{
+    Geometry1 geometry1;
+    Geometry2 geometry2;
+    bgeo::read_wkt(wkt1, geometry1);
+    bgeo::read_wkt(wkt2, geometry2);
+    bgeo::correct(geometry1);
+    bgeo::correct(geometry2);
+
+    Multi result_intersection, result_union;
+    Multi result_intersection_rev, result_union_rev;
+    Multi diff1, diff2, sym_diff;
+
+    bgeo::intersection(geometry1, geometry2, result_intersection);
+    bgeo::intersection(geometry2, geometry1, result_intersection_rev);
+    bgeo::union_(geometry1, geometry2, result_union);
+    bgeo::union_(geometry2, geometry1, result_union_rev);
+    bgeo::difference(geometry1, geometry2, diff1);
+    bgeo::difference(geometry2, geometry1, diff2);
+    bgeo::sym_difference(geometry1, geometry2, sym_diff);
+
+    auto const area1 = bgeo::area(geometry1);
+    auto const area2 = bgeo::area(geometry2);
+    auto const area_union = bgeo::area(result_union);
+    auto const area_intersection = bgeo::area(result_intersection);
+    auto const area_diff1 = bgeo::area(diff1);
+    auto const area_diff2 = bgeo::area(diff2);
+    auto const area_sym_diff = bgeo::area(sym_diff);
+
+    auto const balance = (area1 + area2) - area_union - area_intersection;
+    auto const balance_d1 = (area_union - area2) - area_diff1;
+    auto const balance_d2 = (area_union - area1) - area_diff2;
+    auto const balance_sym = (area_diff1 + area_diff2) - area_sym_diff;
```

Comment:
This math is also used in robustness tests.
It avoids specifying area for every test case.